doc/extending/using_params.txt

```diff
@@ -0,0 +1,140 @@
+.. _extending_op_params:
+
+===============
+Using Op params
+===============
+
+The Op params is a facility to pass some runtime parameters to the
+code of an op without modifying it.  It can enable a single instance
+of C code to serve different needs and therefore reduce compilation.
+
+I will first introduce the parts involved in actually using this
+functionality and then present a simple working example.
+
+The params type
+----------------
+
+You can either reuse an existing type such as :class:`Generic` or
+:class:`CDataType`, or create your own.
+
+Defining a params type
+~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note::
+
+    This section is only relevant if you decide to create your own type.
+
+The first thing you need to do is to define a Theano Type for your
+params object.  It doesn't have to be complete type because only the
+following methods will be used for the type:
+
+  - :meth:`filter <PureType.filter>`
+  - :meth:`__eq__ <PureType.__eq__>`
+  - :meth:`__hash__ <PureType.__hash__>`
+  - :meth:`values_eq <PureType.values_eq>`
+
+Additionaly if you want to use your params with C code, you need the
+following methods:
+
+  - :meth:`c_declare <CLinkerType.c_declare>`
+  - :meth:`c_init <CLinkerType.c_init>`
+  - :meth:`c_extract <CLinkerType.c_extract>`
+  - :meth:`c_cleanup <CLinkerType.c_cleanup>`
+
+You can also define other convinience methods such as
+:meth:`c_headers <CLinkerType.c_headers>` if you need any special things.
+
+
+Registering the params with your Op
+-----------------------------------
+
+To declare that your op uses params you have to set the class
+attribute :attr:`params_type` to an instance of your params Type.
+
+For example if we decide to use a pointer to an int as the params the
+following would be appropriate:
+
+.. code-block:: python
+
+   class MyOp(Op):
+       params_type = CDataType('int')
+
+After that you need to define a :meth:`get_params` method on your
+class with the following signature:
+
+.. code-block:: python
+
+   def get_params(self, node):
+       pass
```

Comment:
Some people are going to read only half of what you wrote, see this and go 'Oh, my method just needs to exist and doesn't need to do anything'. Either give a fake method content consistent with your int example so far or really just show the signature of the method without any implementation.